kotlinx-coroutines-core/wasmWasi/src/internal/CoroutineRunner.kt

```diff
@@ -9,5 +9,6 @@ public fun runTestCoroutine(context: CoroutineContext, block: suspend CoroutineS
     val coroutine = object: AbstractCoroutine<Unit>(newContext, initParentJob = true, active = true) {}
     coroutine.start(CoroutineStart.DEFAULT, coroutine, block)
     runEventLoop()
-    if (coroutine.isCancelled) throw coroutine.getCancellationException().let { it.cause ?: it }
+    coroutine.getCompletionExceptionOrNull()?.let { throw it }
+    check(coroutine.isCompleted) { "Coroutine $coroutine did not complete, but the system reached quiescence" }
```

Comment:
Eh, no, it's better to do it in the opposite order. `getCompletionExceptionOrNull` itself throws exceptions when the coroutine is still active, but really obscure ones.